stories/core/Tree.stories.tsx

```diff
@@ -0,0 +1,404 @@
+  const onSelectedNodeChange = React.useCallback(
+    (nodeId: string, isSelected: boolean) => {
+      if (isSelected) {
+        setSelectedNodes((oldSelected) => [...oldSelected, nodeId]);
+        action(`Selected node ${nodeId}`)();
+      } else {
+        setSelectedNodes((oldSelected) =>
+          oldSelected.filter((item) => item != nodeId),
+        );
+        action(`Unselected node ${nodeId}`)();
+      }
+    },
+    [],
+  );
+
+  const [expandedNodes, setExpandedNodes] = React.useState([
+    'Node-2',
+    'Node-2-1',
+    'Node-3',
+  ]);
+  const onNodeExpanded = React.useCallback(
+    (nodeId: string, isExpanded: boolean) => {
+      if (isExpanded) {
+        setExpandedNodes((oldExpanded) => [...oldExpanded, nodeId]);
+        action(`Expanded node ${nodeId}`)();
+      } else {
+        setExpandedNodes((oldExpanded) =>
+          oldExpanded.filter((item) => item != nodeId),
+        );
+        action(`Closed node ${nodeId}`)();
+      }
+    },
+    [],
+  );
+
+  const [disabledNodes] = React.useState([
+    'Node-4',
+    'Node-3-0',
+    'Node-6',
+    'Node-10',
+  ]);
+
+  const generateItem = React.useCallback(
+    (index: number, parentNode = '', depth = 0): StoryData => {
+      const keyValue = parentNode ? `${parentNode}-${index}` : `${index}`;
+      return {
+        id: `Node-${keyValue}`,
+        label: `Node ${keyValue}`,
+        sublabel: `Sublabel for Node ${keyValue}`,
+        subItems:
+          depth < 10
+            ? Array(Math.round(index % 5))
+                .fill(null)
+                .map((_, index) => generateItem(index, keyValue, depth + 1))
+            : [],
+      };
+    },
+    [],
+  );
+
+  const data = React.useMemo(
+    () =>
+      Array(50)
+        .fill(null)
+        .map((_, index) => generateItem(index)),
+    [generateItem],
+  );
+
+  const getNode = React.useCallback(
+    (node: StoryData): NodeData<StoryData> => {
+      return {
+        subNodes: node.subItems,
+        nodeId: node.id,
+        node: node,
+        isExpanded: expandedNodes.some((id) => id === node.id),
+        isDisabled: disabledNodes.some((id) => id === node.id),
+        isSelected: selectedNodes.some((id) => id === node.id),
+        hasSubNodes: node.subItems.length > 0,
+      };
+    },
+    [disabledNodes, expandedNodes, selectedNodes],
+  );
+
+  return (
+    <Tree<StoryData>
+      data={data}
+      getNode={getNode}
+      nodeRenderer={React.useCallback(
+        ({ node, ...rest }) => (
+          <TreeNode
+            label={node.label}
+            sublabel={node.sublabel}
+            onExpanded={onNodeExpanded}
+            onSelected={onSelectedNodeChange}
+            checkbox={<Checkbox variant='eyeball' disabled={rest.isDisabled} />}
+            icon={<SvgPlaceholder />}
+            {...rest}
+          />
+        ),
+        [onNodeExpanded, onSelectedNodeChange],
+      )}
+    />
+  );
+};
+
+export const AsyncLoading: Story<TreeProps<unknown>> = () => {
```

Comment:
I am thinking to remove this story and bring it back when we will have proper skeleton or other example.